src/shared/util/attributeLogRecordProcessor.ts

```diff
@@ -0,0 +1,14 @@
+import { BatchLogRecordProcessor, LogRecord, LogRecordExporter } from "@opentelemetry/sdk-logs";
+
+export class AttributeLogProcessor extends BatchLogRecordProcessor {
+    private _attributes: { [key: string]: string };
+    constructor(exporter: LogRecordExporter, attributes: { [key: string]: string }) {
+        super(exporter);
+        this._attributes = attributes;
+    }
+    // Override onEmit to apply log record attributes before exporting
+    onEmit(record: LogRecord) {
```

Comment:
Same here, you need to implement the interface LogRecordProcessor not extending and overriding Batch one, then you need to add the LogProcessor using global LoggerProvider